src/molecules/Select/Select.test.tsx

```diff
@@ -1,111 +1,161 @@
-        placeholder="I am placeholder"
-        data-qaid="testQaId"
-      >
-        {({ optionClick }: any) =>
-          citiesData.map((cityData, index) => (
-            <Option key={index} value={cityData.value} onClick={optionClick} />
-          ))
-        }
-      </Select>
-    </>
+const setup = (extraProps?: any) =>
+  renderWithTheme(
+    <Select<Cities>
+      data-qaid="select"
+      options={cities}
+      placeholder="Select City"
+      getOptionLabel={o => o.title}
+      {...extraProps}
+    />
   );
 
-describe('Select', () => {
+describe('<Select />', () => {
   it('renders correctly', () => {
-    expect(setup()).toMatchSnapshot();
+    const { queryByTestId } = setup();
+    expect(queryByTestId('select')).toBeInTheDocument();
+    expect(queryByTestId('select-menu')).not.toBeInTheDocument();
+  });
+
+  it('renders correctly with a defaultValue', () => {
+    const { queryByTestId } = setup({
+      defaultValue: cities[2]
+    });
+
+    expect(queryByTestId('select-input')).toHaveValue('Bristol');
+  });
+
+  it('renders correctly with a custom left hand side icon', () => {
+    const { queryByTestId } = setup({
+      renderLeftIcon: <Icon name="helpCircleOutline" />
+    });
+
+    expect(queryByTestId('select-left-icon')).toBeInTheDocument();
+  });
+
+  it('opens the select when clicking the input', () => {
+    const { queryByTestId } = setup();
+
+    fireEvent.click(queryByTestId('select')!);
+
+    expect(queryByTestId('select-menu')).toBeInTheDocument();
+  });
+
+  it('closes the select when clicking the input', () => {
+    const { queryByTestId } = setup();
+
+    fireEvent.click(queryByTestId('select')!);
+    expect(queryByTestId('select-menu')).toBeInTheDocument();
+
+    fireEvent.click(queryByTestId('select')!);
+    expect(queryByTestId('select-menu')).not.toBeInTheDocument();
   });
 
-  it('renders a placeholder', () => {
-    let wrapper = setup();
-    let placeholder = wrapper.find('input').props().placeholder;
-    expect(placeholder).toBe('I am placeholder');
+  it('closes the select when clicking outside of the input', () => {
+    const { queryByTestId } = setup();
+
+    fireEvent.click(queryByTestId('select')!);
+    expect(queryByTestId('select-menu')).toBeInTheDocument();
+
+    fireEvent.click(document);
+    expect(queryByTestId('select-menu')).not.toBeInTheDocument();
   });
 
-  it('opens the menu when the button is clicked', () => {
-    let wrapper = setup();
-    let button = wrapper.find('button');
-    let options = wrapper.find(Option);
-    expect(options.length).toBe(0);
-    button.simulate('click');
-    let optionUpdated = wrapper.find(Option);
-    expect(optionUpdated.length).toBe(3);
+  it('closes the select when hitting escape', () => {
+    const { queryByTestId } = setup();
+
+    fireEvent.click(queryByTestId('select')!);
+    expect(queryByTestId('select-menu')).toBeInTheDocument();
+
+    fireEvent.keyDown(document, { key: 'Escape', code: 27 });
```

Comment:
I can look it up, but what is the purpose of passing `document` here?

Reply:
dunno, just copied from the docs :D